pex/pex.py

```diff
@@ -413,7 +413,19 @@ def patch_all(path, path_importer_cache, modules):
 
         if self._vars.PEX_EXTRA_SYS_PATH:
             TRACER.log("Adding %s to sys.path" % self._vars.PEX_EXTRA_SYS_PATH)
-            new_sys_path.extend(self._vars.PEX_EXTRA_SYS_PATH.split(":"))
+            extra_sys_path = self._vars.PEX_EXTRA_SYS_PATH.split(":")
+            new_sys_path.extend(extra_sys_path)
+
+            # Let Python subprocesses see the same sys.path additions we see. If those Python
+            # subprocesses are PEX subprocesses, they'll do their own (re-)scrubbing as needed.
+            if inherit_path is InheritPath.FALSE:
+                pythonpath_entries = extra_sys_path
+            else:
+                raw_pythonpath = os.environ.get(self._PYTHONPATH)
+                pythonpath = raw_pythonpath.split(os.pathsep) if raw_pythonpath else []
+                pythonpath_entries = pythonpath + extra_sys_path
+            os.environ[self._PYTHONPATH] = os.pathsep.join(pythonpath_entries)
```

Comment:
This mechanism is broken for some use cases. The fundamental issue is the mechanism of `PEX_EXTRA_SYS_PATH` propagation (`PYTHONPATH`) is not faithful. `PEX_EXTRA_SYS_PATH` appends to the `sys.path` but `PYTHONPATH` prepends. This will need to be reverted or re-worked to produce a faithful representation.